config/vufind/AccountMenu.yaml

```diff
@@ -1,89 +1,117 @@
+      label: Loan History
+      route: checkouts-history
+      icon: user-loan-history
+      checkMethod: checkHistoricloans
 
-  - name: holds
-    label: Holds and Recalls
-    route: holds-list
-    icon: user-holds
-    status: true
-    checkMethod: checkHolds
+    - name: holds
+      label: Holds and Recalls
+      route: holds-list
+      icon: user-holds
+      status: true
+      checkMethod: checkHolds
 
-  - name: storageRetrievalRequests
-    label: Storage Retrieval Requests
-    route: myresearch-storageretrievalrequests
-    icon: user-storage-retrievals
-    status: true
-    checkMethod: checkStorageRetrievalRequests
+    - name: storageRetrievalRequests
+      label: Storage Retrieval Requests
+      route: myresearch-storageretrievalrequests
+      icon: user-storage-retrievals
+      status: true
+      checkMethod: checkStorageRetrievalRequests
 
-  - name: ILLRequests
-    label: Interlibrary Loan Requests
-    route: myresearch-illrequests
-    icon: user-ill-requests
-    status: true
-    checkMethod: checkILLRequests
+    - name: ILLRequests
+      label: Interlibrary Loan Requests
+      route: myresearch-illrequests
+      icon: user-ill-requests
+      status: true
+      checkMethod: checkILLRequests
 
-  - name: fines
-    label: Fines
-    route: myresearch-fines
-    status: true
-    checkMethod: checkFines
-    iconMethod: finesIcon
+    - name: fines
+      label: Fines
+      route: myresearch-fines
+      status: true
+      checkMethod: checkFines
+      iconMethod: finesIcon
 
-  - name: profile
-    label: Profile
-    route: myresearch-profile
-    icon: profile
+    - name: profile
+      label: Profile
+      route: myresearch-profile
+      icon: profile
 
-  - name: librarycards
-    label: Library Cards
-    route: librarycards-home
-    icon: barcode
-    checkMethod: checkLibraryCards
+    - name: librarycards
+      label: Library Cards
+      route: librarycards-home
+      icon: barcode
+      checkMethod: checkLibraryCards
 
-  - name: dgcontent
-    label: Overdrive Content
-    route: overdrive-mycontent
-    icon: overdrive
-    checkMethod: checkOverdrive
+    - name: dgcontent
+      label: Overdrive Content
+      route: overdrive-mycontent
+      icon: overdrive
+      checkMethod: checkOverdrive
 
-  - name: history
-    label: Search History
-    route: search-history
-    icon: search
-    checkMethod: checkHistory
+    - name: history
+      label: Search History
+      route: search-history
+      icon: search
+      checkMethod: checkHistory
 
-  - name: logout
-    label: Log Out
-    route: myresearch-logout
-    icon: sign-out
-    checkMethod: checkLogout
+    - name: logout
+      label: Log Out
+      route: myresearch-logout
+      icon: sign-out
+      checkMethod: checkLogout
+
+Lists:
+  label: Your Lists
+  id: acc-menu-lists-header
+  MenuItems:
+    - template: myresearch/menu-mylists.phtml
+      icon: user-list
+      checkMethod: checkUserlistMode
```

Comment:
Might it make sense to support checkMethod at the group as well as item level? You could probably save some time and redundant checking this way.

Reply:
Added checkMethod for menu groups.

Reply:
Now that you've added the function, do you want to refactor the configuration?

Reply:
Changed.